hilti/include/ast/scope-lookup.h

```diff
@@ -58,15 +57,32 @@ Result<std::pair<NodeRef, ID>> lookupID(const ID& id, const visitor::Position<No
             // Pass back error.
             return std::move(resolved);
 
+        // If the type has the NoInheritScope flag, we skip everything else
+        // in remainder of the path except for the top-level module, to which
+        // we then jump directly. One exception: If the type is part of a
+        // type declaration, we need to check the declaration's scope still
+        // as well.
```

Comment:
I am not sure I understand the last sentence here; were does this happen in the code below?

Reply:
adding another note to the comment, it's what the if/else does